Item/Item.module.less

```diff
@@ -16,13 +40,70 @@
 		line-height: @agate-item-line-height;
 		color: @agate-item-color;
 
+		.itemContent {
+			.label   {
+				color: @agate-item-label-color;
+				font-size: @agate-item-label-font-size;
+			}
+
+			&.hasLabel {
+				&.labelBefore .label {
+					.padding-start-end(0, @agate-item-label-padding);
+				}
+
+				&.labelAfter .label {
+					.padding-start-end(@agate-item-label-padding, 0);
+				}
+
+
+				&.labelAfter,
+				&.labelBefore {
+					.content {
+						box-sizing: border-box;
+						// when showing content and label inline, favor content
+						min-width: @agate-item-content-min-width;
+					}
+
+					.label {
+						box-sizing: border-box;
+						max-width: @agate-item-label-max-width;
+					}
+				}
+			}
+		}
+
+		&.selected {
+			color: @agate-item-selected-color;
+		}
+
 		&.inline {
 			max-width: 240px;
+
+			.itemContent {
+				width: unset;
+
+				&.labelAbove,
```

Comment:
I've added a guard above for `min-width`/`max-width` which allows us to remove this block since those props won't be set for above/below labels.

Reply:
Typically what I've done in the past for these sorts of things is include a class on the base component for this "orientation" scenario and get the checking logic out of javascript, then only apply the min/max-width rules to the horizontal state and min/max-height to the vertical state. This means there are no overrides and no logic in JS determining which is ok.